backend/middlewares/validators/courseValidators.ts

```diff
@@ -0,0 +1,42 @@
+import { Request, Response, NextFunction } from "express";
+import {
+  getApiValidationError,
+  validateArray,
+  validatePrimitive,
+} from "./util";
+import { getErrorMessage } from "../../utilities/errorUtils";
+
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+/* eslint-disable-next-line import/prefer-default-export */
+export const courseRequestDtoValidator = async (
+  req: Request,
+  res: Response,
+  next: NextFunction,
+) => {
+  let body;
+  try {
+    body = JSON.parse(req.body.body);
+  } catch (e: unknown) {
+    return res.status(400).send(getErrorMessage(e));
+  }
+  if (!validatePrimitive(body.title, "string")) {
+    return res.status(400).send(getApiValidationError("title", "string"));
+  }
+  if (!validatePrimitive(body.description, "string")) {
+    return res.status(400).send(getApiValidationError("description", "string"));
+  }
+  if (!validatePrimitive(body.image, "string")) {
+    return res.status(400).send(getApiValidationError("image", "string"));
+  }
+  if (!validatePrimitive(body.previewImage, "string")) {
+    return res
+      .status(400)
+      .send(getApiValidationError("previewImage", "string"));
+  }
+  if (!validateArray(body.lessons, "string")) {
```

Comment:
Validation would probably need to be updated to an array of objects for lessons, doesn't look like util.ts has a validator for custom objects though so maybe we can just drop validation on lessons or write our own validation function.